### PR TITLE
REP-4873 Adding Docker networking to firewall

### PR DESCRIPTION
**Pull request by dmnjohns**
So I tried setting the docker0 bridge to a static address, but did not succeed. All the things I tried ended up breaking Docker or the Docker Puppet module in some way. I'll probably keep poking at this trying to figure it out.

With that said, I believe that the default settings Docker uses are static, and will only change if:
- Docker updates and changes the default settings or behavior
- The IP Docker tries to use by default is already in use

As a result, what I have right now is working. It just sets the firewall settings to work with the default docker0 bridge.

### CODE REVIEW

File: manifests/site.pp [hunk 1]
```diff
@@ -2,6 +2,9 @@
 resources { "firewall":
     purge => true
 }
+# resources { 'firewallchain':
+#     purge => true,
+# }
```
Comment (dmnjohns) [inline manifests/site.pp]: So apparently the `firewall` resource doesn't clear other chain. Clearing other chain isn't strictly necessary, but if we want to, the `firewallchain` resource will do that.

Except that uncommenting this broke something. I didn't pursue the issue too far.

Comment (dmnjohns) [inline manifests/site.pp]: Apparently this was an issue:
https://tickets.puppetlabs.com/browse/MODULES-2769

Comment (dmnjohns) [inline manifests/site.pp]: Apparently even after that, the `firewallchain` resource tries to purge internal chains. What the heck. I think this one might stay commented out.

Comment (dmnjohns) [inline manifests/site.pp]: Looking across GitHub, it seems that others have this problem too:

- https://github.com/dekpient/vagrant-readthedocs/blob/b8d045881c518e3ce9c77a18e19923633b40e531/local_modules/effing_firewall/manifests/init.pp#L17-L19

- https://github.com/shearn89/puppet-toughen/blob/9631d5eaebd42de23ab75f1366763a2d3ba05625/manifests/firewall.pp#L7-L9

**Comment by adrianjgeorge** [comment]
Those sure are some firewall rules. they work?

**Comment by dmnjohns** [comment]
They do.

They are _exactly_ the same rules that Docker was generating. I figured that was the safest approach.

**Comment by adrianjgeorge** [comment]
Legit

**Pull request merged**